src/apps/mailing/services.py

```diff
@@ -38,7 +56,10 @@ def __init__(self) -> None:
         self._initialized = True
 
     async def send(self, message: MessageSchema) -> None:
-        fm = FastMail(self._connection)
+        mailing_class = FastMail
+        if settings.env == "testing":
+            mailing_class = TestMail
+        fm = mailing_class(self._connection)
         await fm.send_message(message)
```

Comment:
If you wanna use different classes for different cases I think you have to take a look the design pattern called Strategy. It will be more suitable for you needs :)